src/dbnode/ts/segment.go

```diff
@@ -123,3 +124,34 @@ func (s *Segment) Finalize() {
 	}
 	s.Tail = nil
 }
+
+// Clone will create a copy of this segment with an optional bytes pool.
+func (s *Segment) Clone(pool pool.CheckedBytesPool) Segment {
+	var (
+		checkedHead, checkedTail checked.Bytes
+	)
+
+	head := s.Head.Bytes()
+	ch := make([]byte, len(head))
+	copy(ch, head)
+	tail := s.Tail.Bytes()
+	ct := make([]byte, len(tail))
+	copy(ct, tail)
+
+	if pool != nil {
+		checkedHead = pool.Get(len(ch))
+		checkedHead.IncRef()
+		checkedHead.Reset(ch)
+		checkedHead.DecRef()
+
+		checkedTail = pool.Get(len(ct))
+		checkedTail.IncRef()
+		checkedTail.Reset(ct)
+		checkedTail.DecRef()
+	} else {
+		checkedHead = checked.NewBytes(ch, nil)
+		checkedTail = checked.NewBytes(ct, nil)
+	}
+
+	return NewSegment(checkedHead, checkedTail, s.Flags)
```

Comment:
You'll want to return FinalizeHead&FinalizeTail since we copy both of them here.

Reply:
Nice, good point.